src/util/github-installations-helper.ts

```diff
@@ -0,0 +1,126 @@
+import { AppInstallation, FailedAppInstallation } from "config/interfaces";
+import { Request } from "express";
+import Logger from "bunyan";
+import { Subscription, SyncStatus } from "models/subscription";
+import { createAppClient } from "utils/get-github-client-config";
+import { getCloudOrServerFromGitHubAppId } from "utils/get-cloud-or-server";
+import { RepoSyncState } from "models/reposyncstate";
+import { statsd } from "config/statsd";
+import { metricError } from "config/metric-names";
+import { groupBy, countBy } from "lodash";
+
+interface FailedConnection {
+	id: number;
+	deleted: boolean;
+	orgName?: string;
+}
+export interface InstallationResults {
+	fulfilled: AppInstallation[];
+	rejected: FailedAppInstallation[];
+	total: number;
+}
+interface SuccessfulConnection extends AppInstallation {
+	isGlobalInstall: boolean;
+}
+interface ConnectionsAndInstallations extends GitHubCloudObj {
+	installations: InstallationResults
+}
+interface GitHubCloudObj {
+	successfulConnections: SuccessfulConnection[]
+	failedConnections: FailedConnection[]
+}
+
+export type ConnectionSyncStatus = "IN PROGRESS" | "FINISHED" | "PENDING" | "FAILED" | undefined;
+const mapSyncStatus = (syncStatus: SyncStatus = SyncStatus.PENDING): ConnectionSyncStatus => {
+	switch (syncStatus) {
+		case "ACTIVE":
+			return "IN PROGRESS";
+		case "COMPLETE":
+			return "FINISHED";
+		default:
+			return syncStatus as ConnectionSyncStatus;
+	}
+};
+
+export const getInstallations = async (subscriptions: Subscription[], log: Logger, gitHubAppId: number | undefined): Promise<InstallationResults> => {
+	const installations = await Promise.allSettled(subscriptions.map((sub) => getInstallation(sub, gitHubAppId, log)));
+	// Had to add "unknown" in between type as lodash types is incorrect for
```

Comment:
/nit is this sentence complete?

Reply:
It was like this from ages ago. But I've updated it.